### ISSUE TITLE
text as dark gray or black for accessibility

### ISSUE THREAD

**Issue by Nick-Levinson** [h5bp/html5-boilerplate#1714]
Contrast is an issue for accessibility and is sometimes cut close, but if a designer thinks text is black when it's near but not black, the option for black should be clearer. I'm tempted to assume that designers who read CSS files usually recognize that #222 is not black, but maybe many don't. In that case, I still wouldn't change the rule in main.css but would add a commented-out style rule setting text to black next to the existing rule for the html element and suggest in a comment that commenting can be edited. (Two of us discussed the issue at <http://www.accessifyforum.com/viewtopic.php?t=34639>.)

**Comment by drublic** [comment]
In my opinion `#222` and `#fff` have some really high contrast. Sure, you can improve it in favor of removing some more "designy" default but the improvement here wouldn't be that big.
At least that's what I think.

**Comment by Nick-Levinson** [comment]
I agree that it's relatively small. My concern was only with text being applied to a background that is too close to #222 for sufficient contrast for visual accessibility. If the background is white, while I haven't checked, I assume #222 gives plenty of contrast. I wouldn't remove #222 from CSS; I'd only remind authors to choose for sufficiently accessible contrast, usually #222 but sometimes #000, by offering the latter commented out with an explanation.

**Comment by roylodder** [comment]
I'm just leaving this here... http://jxnblk.com/colorable/demos/text/?background=%23ffffff&foreground=%23222222

**Comment by drublic** [comment]
@Nick-Levinson I understand this concern. I thought about a note about it in our docs. Maybe you can find a good point where to put it and open a pull request about it?

**Comment by Nick-Levinson** [comment]
I tried to set up a pull request but perhaps I don't have the authority to do so, or it might take me a few hours to navigate all of the processes (creating a fork, etc.), so I'll post here and hope this doesn't inconvenience anyone.

Either of these two solutions should work.

In <css.md>, in the subsection Useful Defaults, at the end of the subsection, add as a paragraph:

For accessibility, if you are using a background other than all-white (including an image behind text), and if the contrast with dark gray text is not enough, in main.css edit the section Base Styles: Opinionated Defaults by editing the html element's style value for color from "#222" (which is darkgray) to "#000" (which is black) (without quotation marks).

The next solution would be near the relevant code, so I prefer this one:

In <main.css>, in the section Base Styles: Opinionated Defaults, before the first item, add:
/*
 * While darkgray is generally good for text and is accessible,
 * if your background (including an image behind text) is other
 * than white, assuring enough contrast for accessibility may
 * require that text be black. If so, decomment the following 
 * style rule (specifying "#000" or black) and comment out the
 * style rule after it (specifying "#222" or darkgray). Do not
 * decomment both.
 *
 * html {
 *     color: #000;
 *     font-size: 1em;
 *     line-height: 1.4;
 * }
 *
 */

**Comment by ramasilveyra** [comment]
The background and color contrast ratio approves WCAG 2.0 AA rules even AAA. http://www.w3.org/TR/WCAG20/#visual-audio-contrast-contrast The minimun ratio is 4.5:1 and Enhanced minimun is 7:1.

With `#222` and `#ffff` we get **15.9:1** http://leaverou.github.io/contrast-ratio/#%23222-on-%23fff

I think we should accept the WCAG rules and don't change it.
